static/usr/lib/systemd/system/tee-supplicant.service.d/core-override.conf

```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: BSD-2-Clause
+# Adapted from https://github.com/OP-TEE/optee_client/blob/6486773583b5983af8250a47cf07eca938e0e422/tee-supplicant/tee-supplicant%40.service.in
+[Unit]
+DefaultDependencies=no
+BindsTo=dev-teepriv0.device var-lib-optee\x2dclient-data-tee.mount
```

Comment:
Does it really BindsTo the mount rather than Requires?

Would `RequireMountsFor=/var/lib/optee-client/data/tee` not work?